HyperPlay-Gaming/hyperplay-desktop-client
# [FIX] HP Games import

- Fix current logic
- Use Manifest file to get version, platform and exe path
- Check for subfolders if dev folders are selected
- Update manifest after import with new info
- Show error messages if invalid folders or EXE not found.


## How to test

**IMPORTANT: Should be tested on all operating systems and for native and non-native games (Windows games on macOS and Linux)
It is not possible to import zipped games, only extracted ones.
It is not possible to import Sideloaded games. Only the ones that were previously installed from the HP store.
The game should be added to the library first, it is not possible to install a game that it is not on the library.
The changes affect only HP games, not Epic and GOG which are working fine and have their logic.**


### Testing if Import works
0. Not necessary but can improve testing speed: Get a game you already have installed, go to the game folder, rename it, or copy to a different place. Then, you can go ahead and uninstall the game from HP. 
1. Go to the install modal and click IMPORT. 
2. Select the Dev folder (the folder that contains the game folder and not the game files).  The game should be installed now and works fine after clicking Select.
3. Repeat Step 1 but now select the Game folder (the one with the game files that is inside the dev folder). The game should be installed and works fine after selecting it.


### Testing the error messages

1. Invalid folder:
     * For any non-installed game, click Install > Import. Select any folder on your machine that does not contain the game. For instance: C:\username\Picures.
     *  An error should be shown saying that the folder is invalid.
   
2. Missing executables:
      * Select the proper game folder, but the game executable is missing
      * An error will show that it is not possible to import since the game files are missing


---

Use the following Checklist if you have changed something on the Backend or Frontend:
